include/SampleLoader.h

```diff
@@ -22,27 +23,66 @@
  *
  */
 
-#ifndef LMMS_GUI_SAMPLE_LOADER_H
-#define LMMS_GUI_SAMPLE_LOADER_H
+#ifndef LMMS_SAMPLE_LOADER_H
+#define LMMS_SAMPLE_LOADER_H
 
+#include <QFileSystemWatcher>
+#include <QObject>
 #include <QString>
-#include <memory>
+#include <map>
 
 #include "SampleBuffer.h"
 #include "lmms_export.h"
 
-namespace lmms::gui {
-class LMMS_EXPORT SampleLoader
+namespace lmms {
+
+class LMMS_EXPORT SampleLoader : public QObject
 {
+	Q_OBJECT
 public:
-	static QString openAudioFile(const QString& previousFile = "");
-	static QString openWaveformFile(const QString& previousFile = "");
-	static std::shared_ptr<const SampleBuffer> createBufferFromFile(const QString& filePath);
-	static std::shared_ptr<const SampleBuffer> createBufferFromBase64(
-		const QString& base64, int sampleRate = Engine::audioEngine()->processingSampleRate());
+	~SampleLoader() override = default;
+
+	using Source = SampleBuffer::Source;
+
+	enum class Cache
+	{
+		None,
+		Read,
+		ReadWrite
+	};
```

Comment:
`None` --> No caching
`Read` --> Reads from cache, but doesn't add new `SampleBuffer`s to the cache - leaves cache unchanged
`ReadWrite` --> Full cache support

In practice this may be complicating things for no reason. Maybe it should just be "use the cache? (true/false)".